### PR TITLE
Add get_guard_expr to symbolic_shapes which returns all guards in a single expression

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87665

### CODE REVIEW

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -439,26 +439,26 @@ def evaluate_guards_for_args(self, *args):
         # and wrap_fake_symbolic
         meta_converter = MetaConverter()
         pytree.tree_map_only(torch.Tensor, partial(meta_converter, shape_env=new_env), args)
-        return all(guard.xreplace(new_env.var_to_val) == value for guard, value, _ in self.guards)
+        return all(guard.xreplace(new_env.var_to_val) for guard, _ in self.guards)
```
Comment (ezyang) [inline torch/fx/experimental/symbolic_shapes.py:442]: how come you can drop the value test

**Review by ezyang — APPROVED** [review]
sure

File: torch/fx/experimental/symbolic_shapes.py (hunk 1 above)
Comment (Chillee) [reply torch/fx/experimental/symbolic_shapes.py:442]: I removed values altogether from the guards.

**Review by voznesenskym — APPROVED** [review]

**Comment by voznesenskym** [comment]
Awesome, ty for making this

**Pull request closed without merging**